src/daemon.rs

```diff
@@ -533,8 +597,23 @@ impl Daemon {
         verbose: bool,
     ) -> Result<Value> {
         let mut args = json!([txhash.to_hex(), verbose]);
-        if let Some(blockhash) = blockhash {
-            args.as_array_mut().unwrap().push(json!(blockhash.to_hex()));
+        let mut conn = self.conn.lock().unwrap();
```

Comment:
Making RPC calls under lock makes my hair stand up. The risk of deadlocks is too great IMO and we already had deadlock issues so I'd really, really like to avoid it. (Disclaimer: I think we already have such things and I think they are not good and there's no good reason to repeat such risky design.)

Reply:
I don't know if there is a simple way to avoid that here. I think this has to do with the fact that we shouldn't do that here at the first place (but Core should have).

The other solution I see is to add txindex as field of Daemon struct and use mutable references to Daemon each time we need to get a transaction. I find it quite disruptive...